### PR TITLE
Insert newlines after streaming tool results

### PR DESCRIPTION
**Pull request by jdormit**
This is a fix for #769. It ensures that we insert a `gptel-response-separator` after a tool result comes in in streaming mode. This prevents weird formatting when `gptel-include-tool-results` is `nil` and the tool result came in between two chat responses (see the linked issue for details).

### CODE REVIEW

**Comment by karthink** [comment]
Thanks for the PR.

This is trickier than it appears.  I need to test some edge cases when I can before merging.

**Comment by jdormit** [comment]
Think I've run into one of those edge cases you were talking about @karthink: we should really be inserting the separator before a chat response, rather than after a tool response - otherwise multiple tool calls in a row leads to a bunch of newlines. Not sure if I'll have time to tackle that today but should be able to soon-ish.

File: gptel-curl.el [hunk 1]
```diff
@@ -266,6 +266,9 @@ Optional RAW disables text properties and transformation."
              (set-marker-insertion-type tracking-marker t)
              (plist-put info :tracking-marker tracking-marker))
            (goto-char tracking-marker)
+           (when (plist-get info :last-was-tool-result)
+             (insert gptel-response-separator)
```
Comment (karthink) [inline gptel-curl.el:270]: Here's another edge case.

When you test Claude it usually responds with text ("I'll run this tool for you"), then a tool call, then more text ("The results of the tool are X").

Suppose instead the LLM responds instead with a tool call followed by a response, as Gemini and OpenAI models do.  (No preamble before the tool call.)

Then you get a response that looks like this (with response-prefix "Response: ")
``` emacs-lisp
User: What is the time in Greece?

Response:

The time in Greece is ...
```

The extra `gptel-response-separator` between `Response: ` and the result is from your `:last-was-tool-call` tracking.

**Comment by karthink** [comment]
Also, a similar fix might be needed for non-streaming responses.

**Comment by karthink** [comment]
@jdormit I see a couple of new commits from you here.  Is the edge case from my comment above handled now?

**Comment by karthink** [comment]
@jdormit I'd like to merge this PR but I'm not sure all the edge cases are covered.  If you're still interested in it, please let me know -- we'll need some tests to ensure that they are, and I can work on merging.

Please let me know if you've moved on as well, and I can close the PR.

**Comment by jdormit** [comment]
Hey @karthink, this one very much dropped off my radar. To be honest I'm not sure I'll have the time to take it up again in the near future - feel free to take it over or close it out.
